aea/aea.py

```diff
@@ -255,7 +259,7 @@ def _handle(self, envelope: Envelope) -> None:
             return
 
         for handler in handlers:
-            self._handle_message_with_handler(msg, handler)
+            self._handle_message_with_handler(deepcopy(msg), handler)
```

Comment:
@marcofavorito we had previously not done that; we need to in order to avoid the same object being accessed by different skills. Do you agree?

Reply:
No, I think we can save the time required to copy the object many times. It could slow down the handling phase due to several copies. Moreover, the public APIs of the objects are already read-only because we use the `@property` annotation.

The message is Python has reference semantics regarding argument passing, Python developers should know that and avoid side effects on the message instance. If we don't want that, we should change the programming model.

Reply:
Ok, so revert this one then.